src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -18629,5 +18861,1539 @@ public static void Main()
+                        Interlocked.Increment(ref FireCount28);
+                        Assert.Equal("B..ctor([System.Int32 Y = 1])", context.ContainingSymbol.ToTestDisplayString());
+                        break;
+                    case "([Attr3]int Z = 4)":
+                        Interlocked.Increment(ref FireCount29);
+                        Assert.Equal("C..ctor([System.Int32 Z = 4])", context.ContainingSymbol.ToTestDisplayString());
+                        break;
+                    case "()":
+                        break;
+                    default:
+                        Assert.True(false);
+                        break;
+                }
+            }
+
+            protected void Handle10(SyntaxNodeAnalysisContext context)
+            {
+                var argumentList = (ArgumentListSyntax)context.Node;
+
+                switch (argumentList.ToString())
+                {
+                    case "(2)":
+                        Interlocked.Increment(ref FireCount30);
+                        Assert.Equal("B..ctor([System.Int32 Y = 1])", context.ContainingSymbol.ToTestDisplayString());
+                        break;
+                    case "(5)":
+                        Interlocked.Increment(ref FireCount31);
+                        Assert.Equal("C..ctor([System.Int32 Z = 4])", context.ContainingSymbol.ToTestDisplayString());
+                        break;
+                    default:
+                        Assert.True(false);
+                        break;
+                }
+            }
+        }
+
+        [Fact]
+        public void AnalyzerActions_02()
+        {
+            var text1 = @"
+record A(int X = 0)
+{}
+
+record C
+{
+    C(int Z = 4)
+    {}
+}
+";
+
+            var analyzer = new AnalyzerActions_02_Analyzer();
+            var comp = CreateCompilation(text1);
+            comp.GetAnalyzerDiagnostics(new[] { analyzer }, null).Verify();
+
+            Assert.Equal(1, analyzer.FireCount1);
+            Assert.Equal(1, analyzer.FireCount2);
+            Assert.Equal(1, analyzer.FireCount3);
+            Assert.Equal(1, analyzer.FireCount4);
+            Assert.Equal(1, analyzer.FireCount5);
+        }
+
+        private class AnalyzerActions_02_Analyzer : DiagnosticAnalyzer
+        {
+            public int FireCount1;
+            public int FireCount2;
+            public int FireCount3;
+            public int FireCount4;
+            public int FireCount5;
+
+            private static readonly DiagnosticDescriptor Descriptor =
+               new DiagnosticDescriptor("XY0000", "Test", "Test", "Test", DiagnosticSeverity.Warning, true, "Test", "Test");
+
+            public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            => ImmutableArray.Create(Descriptor);
+
+            public override void Initialize(AnalysisContext context)
+            {
+                context.RegisterSymbolAction(Handle, SymbolKind.Method);
```

Comment:
Should we also add `context.RegisterSymbolAction(Handle, SymbolKind.NamedType);` to verify callbacks for the record type itself?

Reply:
> Should we also add context.RegisterSymbolAction(Handle, SymbolKind.NamedType); to verify callbacks for the record type itself?

I can certainly add one, I don't think there is anything interesting about the type itself, the PR is was mostly focused on record members.

---
In reply to: [461193457](https://github.com/dotnet/roslyn/pull/46308#discussion_r461193457) [](ancestors = 461193457)